myia/compile/backends/python/python.py

```diff
@@ -511,7 +629,12 @@ def convert_default(self, v):
         # (e.g. for inputs of operations.resolve apply nodes).
         from myia.utils.misc import Namespace
 
-        if isinstance(v, (str, Namespace, Operation)):
+        if isinstance(v, str):
+            # Return a ConstantString so that given string will be inlined.
+            # Given string is wrapped inside 2 double quotes.
+            assert '"' not in v, v
+            return ConstantString(f'"{v}"')
```

Comment:
You can use `repr(v)` to quote the string properly even if it contains `"`.

Also I am not entirely sure what the purpose of `ConstantString` is, can't you return the string directly?

Reply:
Hi ! Ok for `repr` !

About `ConstantString`:

As I currently compile by generating code string, each constant node may be converted to a string.

For example, `my_node = (1, 2, 3)`. I will save `(1, 2, 3)` as a string code,

But I can not inline all constant nodes (e.g. constant tuple nodes), so I need a way to make distinction between constant nodes than must be defined (ie. `constant_node_name = constant_node_value`), and constant nodes than can be inlined (ie. use `constant_node_value` everywhere `constant_node_name` appears). As `constant_node_value` may be a string in both cases, I can not just decide to inline all consant nodes converted to strings, so I must separate strings that can be inlined and ones that can not. I use `ConstantString` to do that, by wrapping strings that can be inlined, and then inline any constant node associated to a `ConstantString`.